### DISCUSSION TITLE
How do the main prompt and region control prompts interact?

### DISCUSSION THREAD

**Discussion by halr9000** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#58] [Q&A] [answered]
My tests show that the main prompt is still "seen" during generation. I'm trying to understand how the other networks (lora etc) work in this situation, as my results are all over the place. Should they impact only the region they are mentioned in? Should I put them up top?

**Comment by pkuliyi2015** [comment, marked as answer]
Your prompt in each region will be appended to the main prompt and take effect; The design is to save words for you, because you can write common things in the main prompt. For example, if I want to draw a picture of a beach, I type:

masterpiece, best quality, highres, extremely detailed 8k wallpaper, very detailed, beach, sunlight

in the main prompt. And later in extension's prompt, say that I want to draw a bikini girl, I add a box and type in:

- 1girl, white hair, black bikini, giant breast, slim waist
- and select Type: Foreground, with a feather value 0.3

In another box I want to draw a slightly different girl:

- 1girl, black hair, black bikini, giant breast, slim waist
- and select Type: Foreground, with a feather value 0.3

I have enough VRAM so I add a huge bbox as the background instead of drawing by tiles, and I type in trees, boat, tourists.

The bbox then appears to be:

<img width="541" alt="image" src="https://user-images.githubusercontent.com/59076257/227577049-db71e4ea-078b-4ae1-ba56-be06acd9ea55.png">

For the negative prompt I just add an EasyNegative embedding at the top. 

Then the result is:

![image](https://user-images.githubusercontent.com/59076257/227577359-b38c1868-6f87-47d2-8e8f-3614f4a9805e.png)
